shared-libs/transitions/test/unit/transitions/create_user_for_contacts.js

```diff
@@ -0,0 +1,595 @@
+const sinon = require('sinon');
+const { assert } = require('chai');
+const config = require('../../../src/config');
+const db = require('../../../src/db');
+const transition = require('../../../src/transitions/create_user_for_contacts');
+const { people } = require('@medic/contacts')(config, db);
+const { users } = require('@medic/user-management')(config, db);
+const contactTypeUtils = require('@medic/contact-types-utils');
+
+const deepFreeze = obj => {
+  Object
+    .keys(obj)
+    .filter(prop => typeof obj[prop] === 'object' && !Object.isFrozen(obj[prop]))
+    .forEach(prop => deepFreeze(obj[prop]));
+  return Object.freeze(obj);
+};
+
+const ORIGINAL_CONTACT = deepFreeze({
+  _id: 'original-contact-id', parent: {
+    _id: 'parent-id',
+  }
+});
+
+const NEW_CONTACT = deepFreeze({
+  _id: 'new-contact-id', parent: {
+    _id: 'parent-id',
+  }, phone: '+1234567890', name: 'New Contact',
+});
+
+const ORIGINAL_USER = deepFreeze({
+  _id: 'org.couchdb.user:original-user-id', name: `original-user`, contact: ORIGINAL_CONTACT, roles: ['chw'],
+});
+
+const getReplacedContact = (
+  status,
+  replacement_contact_id = NEW_CONTACT._id,
+) => ({
+  _id: 'replaced-id',
+  parent: {
+    _id: 'parent-id',
+  },
+  user_for_contact: {
+    replace: {
+      [ORIGINAL_USER.name]: {
+        replacement_contact_id,
+        status,
+      }
+    }
+  }
+});
+
+describe('create_user_for_contacts', () => {
+  beforeEach(() => {
+    config.init({
+      getAll: sinon
+        .stub()
+        .returns({}), get: sinon.stub(),
+    });
+  });
+
+  afterEach(() => {
+    sinon.reset();
+    sinon.restore();
+  });
+
+  it('has the correct properties', () => {
+    assert.equal(transition.name, 'create_user_for_contacts');
```

Comment:
Most of our existing `expect` code does not seem to follow our [method chaining style](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/#method-chaining) which would seem to require all expects to be on at least two lines....

Reply:
I'm unsure if you should count the expect assertions as methods.

Reply:
Webstorm does....  :facepalm:

Reply:
Oh, I see! 
Our rule is specifically set for method chaining. I would expect to see something like this an be happy with it: 
```javascript
object.property.subproperty
  .method1()
  .method2()
  .method3();
```

instead of 

```javascript
object
  .property
  .subproperty
  .method1()
  .method2()
  .method3();
```

But I doubt Webstorm will make a difference between the two.

Reply:
So, interestingly, Webstorm can handle this just fine:

```js
object.property.subproperty
  .method1()
  .method2()
  .method3();
```
Webstorm can tell the different between function calls and properties. If you use the `Chained Method Calls: Wrap Always` option in the code style, it will not try to start wrapping until after the _first function call_. The problem is, for something like `expect(transition.name).to.equal('create_user_for_contacts');`, the first function call is `expect(transition.name)` and it will try to wrap everything after that....